File: C#/0724. Find Pivot Index.cs
public class Solution {
    public int PivotIndex(int[] nums) {
        int[] numleft=new int[nums.Length];
        int[] numright=new int[nums.Length];
        int leftsum=0;
        for(int i=0;i<nums.Length;i++){
            numleft[i]=leftsum;
            leftsum+=nums[i];
        }
        int rightsum=0;
        for(int i=nums.Length-1;i>=0;i--){
            numright[i]=rightsum;
            rightsum+=nums[i];
        }
        for (int i=0;i<nums.Length;i++){
            if(numleft[i]==numright[i]){
                return i;
            }
        }
        return -1;
    }
}

File: archives/20190519python/0724. Find Pivot Index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        
        left_sum=0
        right_sum=sum(nums[1:])
        if nums and left_sum==right_sum:
            return 0
        for k in range(1,len(nums)):
            left_sum+=nums[k-1]
            right_sum-=nums[k]
            if left_sum==right_sum:
                return k
        return -1
